src/OpenTelemetry.Extensions.Enrichment/README.md

```diff
@@ -6,6 +6,165 @@
+    {
+        bag.Add("my key", "my value");
+    }
+}
+```
+
+For every `Activity`, the `public override void Enrich(TraceEnrichmentBag bag)`
+method is guaranteed to be called exactly once. Semantically,
+for the example above it means that a new [tag object](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.tagobjects?view=net-7.0)
+with the `my key` key and the `my value` value will be added to every `Activity`
+in your application.
+
+### Step 3: Register enricher class
+
+Add your custom enricher class to the `TracerProviderBuilder` by calling the
+`AddTraceEnricher<T>()` method. Configure `ActivitySource` and an exporter as usual:
+
+```csharp
+using var MyActivitySource = new ActivitySource("MyCompany.MyProduct.MyLibrary");
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource("MyCompany.MyProduct.MyLibrary")
+    .AddTraceEnricher<MyTraceEnricher>()
+    .AddConsoleExporter()
+    .Build();
+```
+
+Alternatively, you can add your custom enricher to the `IServiceCollection`
+(as well as `ActivitySource` and exporter), typically
+this is done inside the [ConfigureServices()](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices)
+method:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddOpenTelemetry().WithTracing((builder) => builder
+        .AddSource("MyCompany.MyProduct.MyLibrary")
+        .AddTraceEnricher<MyTraceEnricher>()
+        .AddConsoleExporter());
+}
+```
+
+> **Note**
+> The `AddTraceEnricher()` method call should be done *before* registering exporter
+related Activity processors.
+
+### Step 4: Usage
+
+Create an `Activity` and add tags as usual:
+
+```csharp
+using var activity = myActivitySource.StartActivity("SayHello");
+activity?.SetTag("hello", "world");
+```
+
+Run your application and verify that the `my key` tag is added to `Activity`:
+
+```shell
+Activity.TraceId:            0e1dc24e2e63796bfc8186e24f916f5f
+Activity.SpanId:             bfcbf9d5746009d6
+Activity.TraceFlags:         Recorded
+Activity.ActivitySourceName: MyCompany.MyProduct.MyLibrary
+Activity.DisplayName:        SayHello
+Activity.Kind:               Internal
+Activity.StartTime:          2023-03-20T09:39:21.9642338Z
+Activity.Duration:           00:00:00.0016887
+Activity.Tags:
+    hello: world
+    my key: my value
+Resource associated with Activity:
+    service.name: unknown_service:Examples.Enrichment
+```
+
+### Extension methods
+
+There are a bunch of extension methods available for registering your custom trace
+enricher class. The methods that utilize `IServiceCollection` are tailored for
+Dependency Injection-based situations, allowing for registration of
+`TracerProviderBuilder` and `IServiceCollection` at different locations
+within your codebase. Conversely, the methods relying on `TracerProviderBuilder`
+are meant to be employed in non-DI scenarios.
```

Comment:
> Conversely, the methods relying on `TracerProviderBuilder` are meant to be employed in non-DI scenarios.

Kind of misleading because both cases are really using DI 😄

I would say the `IServiceCollection` versions are meant for general library authors who may not have a reference to `TracerProviderBuilder` or who want to register enrichers with other general services. `TracerProviderBuilder` versions are for OTel component authors.